frontend/src/context/LocationsContext.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useRef, useState, useCallback } from 'react';
+import { Admin, Location } from '../types';
+import { useReq } from './req';
+
+type locationsState = {
+  locations: Array<Location>;
+  refreshLocations: () => Promise<void>;
+};
+
+const initialState: locationsState = {
+  locations: [],
+  refreshLocations: async () => undefined,
+};
+
+const LocationsContext = React.createContext(initialState);
+export const useLocations = () => React.useContext(LocationsContext);
+
+type locationsProviderProps = {
+  children: React.ReactNode;
+};
+
+export const LocationsProvider = ({ children }: locationsProviderProps) => {
+  const componentMounted = useRef(true);
```

Comment:
dont need componentMounted, i noticed it doesn't always work and sometimes causes issues